src/neptune/common/backends/utils.py

```diff
@@ -95,14 +95,22 @@ def wrapper(*args, **kwargs):
                 HTTPServiceUnavailable,
                 HTTPGatewayTimeout,
                 HTTPBadGateway,
-                HTTPTooManyRequests,
                 HTTPInternalServerError,
                 NewConnectionError,
                 ChunkedEncodingError,
             ) as e:
                 time.sleep(min(2 ** min(10, retry), MAX_RETRY_TIME))
                 last_exception = e
                 continue
+            except HTTPTooManyRequests as e:
+                response_headers = e.response.json()["headers"]
```

Comment:
Are we really really sure that this `headers` exists in On-Prems as well?

Reply:
Changed :D